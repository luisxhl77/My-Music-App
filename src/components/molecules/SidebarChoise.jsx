import styled from "styled-components";

export const SiderbarChoice = ({title,Icon}) => {
    return (
        <ChoiceOptions>
            {/*Si existe Icon agreguelo*/}
            { Icon && <Icon/> } 
            { Icon ? <h4>{title}</h4> : <h5>{title}</h5> }
            {/*Si existe Icon agreguelo de lo contrario no */}
        </ChoiceOptions>
    )
};

const ChoiceOptions = styled.li`
    display: flex;
    align-items: center;
    color: #b2b2b2;
    height: 41px;
    transition: 300ms color ease-in;
    cursor: pointer;
    padding: 0px 0px 0px 5px;
    &:hover{
        color: #fff;
    }
    & h5{
        margin: 10px 0px 0px 5px;
    }
    & .MuiSvgIcon-root{
        font-size: 31px;
        padding: 0px 10px 0px 0px;
    }
`;