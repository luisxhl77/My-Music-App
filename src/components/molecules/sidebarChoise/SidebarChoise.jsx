import { NavLink } from "react-router-dom";
import PropTypes from 'prop-types';
import "./sidebarChoise.scss";

export const SiderbarChoice = ({ title, Icon, nav }) => {

    return (
        <li>
            <NavLink to={nav} className={ ({ isActive }) => `choiceOptions ${ isActive ? 'activeLink' : ''} `}>
                { Icon && <Icon/> }
                { Icon ? <h4 aria-label="title4">{ title }</h4> : <h5 aria-label="title5" className="choiceOptions__title">{ title }</h5> }
            </NavLink>
        </li>
    )

};

SiderbarChoice.propTypes = {
    title: PropTypes.string.isRequired, 
    nav: PropTypes.string.isRequired
}