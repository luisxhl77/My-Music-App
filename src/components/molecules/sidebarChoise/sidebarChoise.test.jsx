import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SiderbarChoice } from "./SidebarChoise";
import { Home } from "@mui/icons-material";

describe('Pruebas de <SiderbarChoise/> ', () => {
    
    const title = "Home" 
    const icon  = Home 
    const nav   = "/home"

    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render( 
            <MemoryRouter>
                <SiderbarChoice/>
            </MemoryRouter>        
        );
        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar el nombre en H4 si tiene un icono', () => {
        
        render( 
            <MemoryRouter>
                <SiderbarChoice
                    title = {title} 
                    Icon  = {icon}
                    nav   = {nav} 
                />
            </MemoryRouter>
        );

        expect( screen.getByText( title )).toBe(screen.getByLabelText('title4'))
    })

    test('debe de mostrar el nombre en H5 si no hay un icono', () => {
        
        render( 
            <MemoryRouter>
                <SiderbarChoice
                    title = {title} 
                    nav   = {nav} 
                />
            </MemoryRouter>
        );

        expect( screen.getByText( title )).toBe(screen.getByLabelText('title5'))
    })

})