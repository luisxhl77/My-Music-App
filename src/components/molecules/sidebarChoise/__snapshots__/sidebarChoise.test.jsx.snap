// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <SiderbarChoise/>  debe de hacer match con el snapshot 1`] = `
<div>
  <li>
    <a
      aria-current="page"
      class="choiceOptions activeLink "
      href="/"
    >
      <h5
        aria-label="title5"
        class="choiceOptions__title"
      />
    </a>
  </li>
</div>
`;
